sw/cheri/tests/hyperram_test.cc

```diff
@@ -0,0 +1,294 @@
+#define CHERIOT_NO_AMBIENT_MALLOC
+#define CHERIOT_NO_NEW_DELETE
+#define CHERIOT_PLATFORM_CUSTOM_UART
+
+#include "../../common/defs.h"
+#include "../common/uart-utils.hh"
+
+#include <cheri.hh>
+#include <platform-uart.hh>
+#include <stdint.h>
+#include <ds/xoroshiro.h>
+
+using namespace CHERI;
+
+const int RandTestBlockSize = 256;
+const int HyperramSize = (1024 * 1024) / 4;
```

Comment:
Note this is intended as a soak test on the real board (i.e. leave it running and see if anything fails, hopefully good for flushing out reliability issues). It has an infinite loop that runs the same tests over and over.

You can run it on Verilator but I'd suggest adjusting these numbers so it runs far fewer tests per loop. E.g. change this to:

```
const int RandTestBlockSize = 16;
const int HyperramSize = 16;
```

Once we have the baremetal testsuite in we can adapt this so we've got a soak mode and a terminating test mode.